autoload/neomake/makers/ft/typescript.vim

```diff
@@ -9,19 +9,14 @@ function! neomake#makers#ft#typescript#tsc() abort
     return {
         \ 'args': ['--project', neomake#utils#FindGlobFile(expand('%:p:h'), 'tsconfig.json'), '--noEmit'],
         \ 'append_file': 0,
-        \ 'errorformat':
-            \ '%E%f %#(%l\,%c): error %m,' .
-            \ '%E%f %#(%l\,%c): %m,' .
-            \ '%Eerror %m,' .
-            \ '%C%\s%\+%m'
+        \ 'errorformat': '%f(%l\,%c): %m'
```

Comment:
There's not much more I can do from here - are you using the `hello.ts` you provided as an example?
I've just tried it again by adding `hello.ts(1,7): error TS2322: Type '"hello"' is not assignable to type 'Number'.` to an empty buffer, then `:let &efm = '%E%f %#(%l\,%c): %m'` and `:cgetbuffer | copen` will display it properly.